src/Uno.UI.BindingHelper.Android/Uno.UI.BindingHelper.Android.netcoremobile.csproj

```diff
@@ -25,7 +25,7 @@
 		<AndroidSdkBuildToolsVersion Condition="'$(AndroidSdkBuildToolsVersion)' == ''">28.0.3</AndroidSdkBuildToolsVersion>
 	</PropertyGroup>
 
-	<ItemGroup Condition="'$(TargetPlatformIdentifier)' == 'android'">
+	<ItemGroup>
```

Comment:
Before this PR, `TargetPlatformIdentifier` was evaluated as `''`, so this condition was always false.

With this PR, the condition evaluates to true, causing:

```
> C:\a\1\s\src\Uno.UI.BindingHelper.Android\Uno.UI.BindingHelper.Android.netcoremobile.csproj error NU1604: Warning As Error: Project dependency Xamarin.AndroidX.AppCompat does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [C:\a\1\s\src\Uno.UI-netcoremobile-only.slnf]
```

The failure happens during `_GenerateProjectRestoreGraphAllFrameworks` so a specific TargetFramework isn't set already.

@jeromelaban I think the safest option is to keep the condition removed as I did, but combine and remove the condition from https://github.com/unoplatform/uno/blob/a61e4badda6147a286aaa9290f1fc874ed65d3e0/src/Directory.Build.targets#L119-L133 so that the version is seen when evaluating the restore graph.